### PR TITLE
Cdm row reader

### PR DESCRIPTION
**Pull request by george-zubrienko**
Port functionality and tests from https://github.com/SneaksAndData/arcane-framework/blob/main/test/Operations/CsvOperationsTests.cs

### CODE REVIEW

File: framework/arcane-framework/src/test/scala/models/CdmParserTests.scala [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.must.Matchers
+import org.scalatest.matchers.should.Matchers.should
+import org.scalatest.prop.TableDrivenPropertyChecks.*
+
+
+class CdmParserTests extends AnyFlatSpec with Matchers {
+
+  private val validCsvLines = Table(
```
Comment (s-vitaliy) [inline framework/arcane-framework/src/test/scala/models/CdmParserTests.scala:14]: It seems that this method does not handle incorrect CSV files [as it did in C#](https://github.com/SneaksAndData/arcane-framework/blob/8ffdc749f044d544605a2bcd7faee8d84b5e59f6/test/Operations/CsvOperationsTests.cs#L15).

File: framework/arcane-framework/src/main/scala/models/cdm/CdmParser.scala [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        (isDelimiter(currentPosition), isEol(currentPosition)) match
+          // empty value at the end of the line
+          case (true, true) if isQuote(currentPosition - 1) => Some(line.slice(from = fromIndex, until = currentPosition - 1)) // if there is a quote at the end, move back by 1 character
+          case (true, true) => Some(line.slice(from = fromIndex, until = currentPosition))
+          case (false, true) if !isQuote(currentPosition) => Some(line.slice(from = fromIndex, until = currentPosition + 1))
+          case (false, true) if isQuote(currentPosition - 1) => Some(line.slice(from = fromIndex, until = currentPosition - 1))
+          case (true, false) if isQuote(currentPosition - 1) => Some(line.slice(from = fromIndex, until = currentPosition - 1))
+          case _ => Some(line.slice(from = fromIndex, until = currentPosition))
+
+    def scanLine(charIndex: Int, quoteSum: Int, prevCharIndex: Int): IndexedSeq[Option[String]] = {
```
Comment (s-vitaliy) [inline framework/arcane-framework/src/main/scala/models/cdm/CdmParser.scala:30]: Is it supposed to be tail recursion? If yes, annotate it with the `@tailrec` annotations, please.

**Review by s-vitaliy — APPROVED** [review]
LGTM in general, left some comments

File: framework/arcane-framework/src/main/scala/models/cdm/CdmParser.scala (hunk 2 above)
Comment (george-zubrienko) [reply framework/arcane-framework/src/main/scala/models/cdm/CdmParser.scala:30]: no this function is not tail-recursive

File: framework/arcane-framework/src/test/scala/models/CdmParserTests.scala (hunk 1 above)
Comment (george-zubrienko) [reply framework/arcane-framework/src/test/scala/models/CdmParserTests.scala:14]: yeah will add before merge

File: framework/arcane-framework/src/main/scala/models/cdm/CdmParser.scala (hunk 2 above)
Comment (s-vitaliy) [reply framework/arcane-framework/src/main/scala/models/cdm/CdmParser.scala:30]: Ok. I have a slight concern about the memory consumption by this function, but we can still test it with real data when we run it in the cluster

**Review by s-vitaliy — APPROVED** [review]

**Pull request merged**